File: code/ch10/10.1.2.ClimbingStairs.cpp
class Solution {
    public:
        int climbStairs(int n) {
            if (n < 2) return n;
            
            int first  = 1;
            int second = 2;

            for (int i = 3; i < n + 1; i++) {
                second = first + second;
                first  = second - first;
            }

            return second;
        }
};

File: code/ch10/10.1.2.ClimbingStairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n < 2:
            return n
        first, second = 1, 2
        for i in range(3, n + 1):
            # third = first + second
            # first = second
            # second = third
            second = first + second
            first = second - first
        return second
